platform/admin/configurator/admin-configurator-impl/src/main/java/org/codice/ddf/admin/configurator/impl/Operation.java

```diff
@@ -0,0 +1,28 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ **/
+package org.codice.ddf.admin.configurator.impl;
+
+import org.codice.ddf.admin.configurator.ConfiguratorException;
+
+/**
+ * <b> This code is experimental. While this interface is functional and tested, it may change or be
+ * removed in a future version of the library. </b>
+ */
+interface Operation<T, S> {
+    T commit() throws ConfiguratorException;
```

Comment:
There seems to be one case where T is not Void and in that case commit always returns null and rollback actually returns a String for ManagedServiceOperation.   Based on that I am not sure if commit and rollback should share the same return.

It might make more sense to have commit return a Result and rollback to return some kind of RollbackResult.  This would also remove the need for create Results and for control flow by exception.  Result and RollbackResult could then store transaction result data.